openfl/transport/grpc/aggregator_client.py

```diff
@@ -245,21 +246,8 @@ def __init__(
         self.aggregator_uuid = aggregator_uuid
         self.federation_uuid = federation_uuid
         self.single_col_cert_common_name = single_col_cert_common_name
-
-        # Adding an interceptor for RPC Errors
-        self.interceptors = (
-            RetryOnRpcErrorClientInterceptor(
-                sleeping_policy=ConstantBackoff(
-                    logger=self.logger,
-                    reconnect_interval=int(kwargs.get("client_reconnect_interval", 1)),
-                    uri=self.uri,
-                ),
-                status_for_retry=(grpc.StatusCode.UNAVAILABLE,),
-            ),
-        )
-        self.stub = aggregator_pb2_grpc.AggregatorStub(
-            grpc.intercept_channel(self.channel, *self.interceptors)
-        )
```

Comment:
What's the reasoning behind removing the this interceptor? I think the gRPC connection can generally benefit from retries (due to temporary network disconnects unrelated to certificates f.e.).

PS : on a side note, wouldn't it be more elegant if you extended the `RetryOnRpcErrorClientInterceptor` to also fetch the certificate?

Reply:
As far as I understand, the interceptor works on  a given channel - however, if we need to fetch new certs we also need to create a new channel and this is why intercepting the call won't work here

Reply:
OK, this is true for the case when you have to re-establish the channel. But I don't understand why you need to remove the retries within an existing channel where the TLS cert hasn't changed?

Reply:
We can have a more fine-tuned error handling and differentiate between errors that are due to a bad TLS handshake (on which we want to refresh the connection) and those that don't require refreshing the certificates.  However, I do think that for the current stage, it is better to take the extreme approach (as we don't have much communication in any case), we can fine-tune it in the future (for example, I saw that on TLS error the error is also UNAVAILABLE, so we can try and refine it but I don't think this should be the focus right now)

Reply:
I think I understand. You suggest the retry to happen at the level of the client's request loop (rather than at the lower gRPC level):
```python
        while True:
            try:
                response = func(self, *args, **kwargs)
                break
```